net/http/authz/authorizer.go

```diff
@@ -0,0 +1,96 @@
+package authz
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/golang/protobuf/proto"
+
+	"chain/database/raft"
+	"chain/errors"
+	"chain/net/http/authn"
+)
+
+const grantPrefix = "/core/grant/"
+
+type Authorizer struct {
+	raftDB        *raft.Service
+	raftPrefix    string
+	policyByRoute map[string][]string
+}
+
+func NewAuthorizer(rdb *raft.Service, prefix string, policyMap map[string][]string) *Authorizer {
+	return &Authorizer{
+		raftDB:        rdb,
+		raftPrefix:    prefix,
+		policyByRoute: policyMap,
+	}
+}
+
+func (a *Authorizer) Authorize(req *http.Request) error {
+	policies := a.policyByRoute[req.RequestURI]
+	if policies == nil {
+		return errors.New("missing policy on this route")
+	}
+
+	grants, err := grantsByPolicies(a.raftDB, policies)
+	if err != nil {
+		return errors.Wrap(err)
+	}
+
+	if !authorized(req.Context(), grants) {
+		return errors.New("not authorized")
+	}
+
+	return nil
+}
+
+func authzToken(ctx context.Context, grants []*Grant) bool {
+	for _, g := range grants {
+		if g.GuardType == "access_token" {
+			if accessTokenGuardData(g) == authn.Token(ctx) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func authzLocalhost(ctx context.Context, grants []*Grant) bool {
+	for _, g := range grants {
+		if g.GuardType == "localhost" {
+			return true
+		}
+	}
+	return authn.Localhost(ctx)
+}
+
+func accessTokenGuardData(grant *Grant) string {
+	var data map[string]string
```

Comment:
Consider using a struct type here and taking advantage of the fact that the value doesn't get set when there's an error:

```go
var v struct{ ID string }
json.Unmarshal(grant.GuardData, &v) // ignore error, returns "" on failure
return v.ID
```